### PR TITLE
Recreate Quttera branch

### PR DESCRIPTION
**Pull request by hayzer**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by YuvHayun** [comment]
Hi @hayzer, thank you for your contribution.
Please fill the contribution form and let me know when the pr is ready for review.

File: Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner_test.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import json
```
Comment (YuvHayun) [inline Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner_test.py:1]: Please add a "given - when - then" documentation.
also, note that you have a few tests that fail.

File: Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+
+''' MAIN FUNCTION '''
+def main(params: dict, args: dict, command: str):
+    """main function, parses params and runs command functions
+    """
+    demisto.debug(f"Command called {command}")
+    demisto.results(f"Args are {args}")
+    demisto.results(f"params are {params}")
+    client = Client(params)
```
Comment (YuvHayun) [inline Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.py:160]: Let's also add a test-module command, 
test-module should test a simple argument less request - just to ensure there's a connection.

**Review by YuvHayun — CHANGES_REQUESTED** [review]
Hey @hayzer, please see my comments below.
Also, please make sure to fill the contribution form.

**Comment by YuvHayun** [comment]
Hi @hayzer, we are now entering the holidays period and therefore we might not be available.
Sorry in advance for any long-delay that'll take us to reply.

**Comment by edik24** [comment]
Hi @hayzer,
any update on the requested changes?

File: Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/quttera-api.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/python
```
[the remark this answers is not in the document]
Comment (hayzer) [reply Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/quttera-api.py]: Removed that file as it only serve the local test

File: Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/code-workspace.code-workspace [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
```
[the remark this answers is not in the document]
Comment (hayzer) [reply Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/code-workspace.code-workspace:1]: Removed as it's not relevant.

File: Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.yml [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+  required: true
+  type: 4
+- display: Quttera Scanner URL
+  name: base_url
+  required: true
+  type: 0
+script:
+  type: python
+  subtype: python3
+  dockerimage: demisto/python3:3.10.4.28442
```
[the remark this answers is not in the document]
Comment (hayzer) [reply Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.yml]: Fixed

File: Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.py [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+    :type client: ``Client`e
+    :param Client: Qutter API client
+
+    :type args: ``Dict[str, Any]``
+    :param args: all command arguments from ``demisto.args()``
+
+    :return: ``CommandResults`` object contain the start scan response
+    :rtype: ``CommandResults``
+    """
+    domain = args['domain']
```
[the remark this answers is not in the document]
Comment (hayzer) [reply Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.py]: Fixed

File: Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner_description.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
+# Quttera Website Malware Scanner
+
+This integration enables the Quttera scanner API
+
+# Credentials
+
+Use API key supply by Quttera
```
[the remark this answers is not in the document]
Comment (hayzer) [reply Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner_description.md]: Fixed

File: Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/code-workspace.code-workspace (hunk 4 above)
[the remark this answers is not in the document]
Comment (YuvHayun) [reply Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/code-workspace.code-workspace:1]: Seems like it wasn't removed.

File: Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.py [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+
+        operations = {
+            "scan": f"/api/v3/{self.apikey}/url/scan",
+            "rescan": f"/api/v3/{self.apikey}/url/scan",
+            "statscan": f"/api/v3/{self.apikey}/url/status",
+            "malware": f"/api/v3/{self.apikey}/url/report",
+            "blacklist": f"/api/v3/{self.apikey}/blacklist/status"
+        }
+
+        op = operations[operation]
```
[the remark this answers is not in the document]
Comment (YuvHayun) [reply Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.py:41]: still need change

File: Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.py [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+    return_results(result)
+
+
+''' MAIN FUNCTION '''
+def main(params: dict, args: dict, command: str):
+    """main function, parses params and runs command functions
+    """
+    demisto.debug(f"Command called {command}")
+    demisto.results(f"Args are {args}")
+    demisto.results(f"params are {params}")
```
[the remark this answers is not in the document]
Comment (YuvHayun) [reply Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.py:159]: Still need change.

**Review by YuvHayun — CHANGES_REQUESTED** [review]
Hey @hayzer, please notice that are few unresolved comments that require your attention.
Also, please pull changes from master to update your repo, and make sure to fill out the contribution form.
Fill free to contact me if you have any further questions.

File: Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/README.md [hunk 10]
```diff
@@ -0,0 +1,22 @@
+Quttera API inspects URLs, domains or websites for security issues.
```
[the remark this answers is not in the document]
Comment (hayzer) [reply Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/README.md]: I've rerun this command multiple time but it has no effect on the README file.

File: Packs/QutteraWebsiteMalwareScanner/pack_metadata.json [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+    ],
+    "useCases": [],
+    "keywords": [],
+    "devEmail": [
+        "hayzer@gmail.com"
+    ],
+    "githubUser": [
+        "hayzer"
+    ]
+}
```
[the remark this answers is not in the document]
Comment (hayzer) [reply Packs/QutteraWebsiteMalwareScanner/pack_metadata.json:36]: Fixed

**Comment by YuvHayun** [comment]
@hayzer let me know when you're ready for re-review.

**Comment by YuvHayun** [comment]
ֿHey @hayzer, still waiting for your comment & fixes.

**Comment by hayzer** [comment]
Hi Yuval,

Yes, I know. I was overwhelmed with other tasks. Next week expects to fix
your comments.

Best regards,
Thomas

**Comment by YuvHayun** [comment]
Hi @hayzer, I see you started working on the pr.
Please let me know when it's ready to re-review.

**Comment by hayzer** [comment]
Hi Yuval,

Indeed, we're making progress at the moment. I will let you know when we're
done.

Best regards,
Thomas

File: Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.py (hunk 2 above)
Comment (hayzer) [reply Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.py:183]: Fixed

**Comment by YuvHayun** [comment]
Hey @hayzer still waiting for your update.

**Comment by hayzer** [comment]
Hi Yuval,

Could you help me with my question first? It seems it's the last thing
before the build is green.

Best regards,
Thomas

File: Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.py [hunk 12]
```diff
@@ -0,0 +1,205 @@
...
+    elif command == 'test-module':
+        results = test_module(client)
+    else:
+        raise NotImplementedError(f'Command {command} not implemented')
+    return_results(results)
+
+
+if __name__ in ('__main__', '__builtin__', 'builtins'):
+    try:
+        main(demisto.params(), demisto.args(), demisto.command())
```
Comment (YuvHayun) [inline Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.py:203]:
```suggestion
        main()
```

File: Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.py [hunk 13]
```diff
@@ -0,0 +1,205 @@
...
+    resp = client._test_module()
+    if resp.get('errorstr') == 'success':
+        return 'ok'
+    raise DemistoException(f'Service is down: ${resp}')
+
+
+def main(params: dict, args: dict, command: str):
+    """main function, parses params and runs command functions
+    """
+    demisto.info(f"Command called {command}")
```
Comment (YuvHayun) [inline Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.py:180]:
```suggestion
def main()  # pragma: no cover
    """main function, parses params and runs command functions
    """
    params = demisto.params()
    args = demisto.args()
    command = demisto.command()
    demisto.info(f"Command called {command}")
```

**Review by YuvHayun — CHANGES_REQUESTED** [review]
@hayzer I added two fixes suggestions, this should help you pass the lint step.

File: Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.py [hunk 14]
```diff
@@ -0,0 +1,208 @@
...
+
+    Return True if request succeeded
+    """
+    resp = client._test_module()
+    if resp.get('errorstr') == 'success':
+        return 'ok'
+    raise DemistoException(f'Service is down: ${resp}')
+
+
+def main()  # pragma: no cover
```
Comment (YuvHayun) [inline Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.py:177]:
```suggestion
def main():  # pragma: no cover
```

My bad, missed the ":"

**Review by YuvHayun — CHANGES_REQUESTED** [review]

File: Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.yml [hunk 15]
```diff
@@ -0,0 +1,156 @@
...
+  required: true
+  type: 4
+- display: Quttera Scanner URL
+  name: base_url
+  required: true
+  type: 0
+script:
+  type: python
+  subtype: python3
+  dockerimage: demisto/python3:3.10.8.39276
```
Comment (YuvHayun) [inline Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.yml:25]:
```suggestion
  dockerimage: demisto/python3:3.10.9.40422
```

File: Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.py [hunk 16]
```diff
@@ -0,0 +1,208 @@
...
+    else:
+        raise NotImplementedError(f'Command {command} not implemented')
+    return_results(results)
+
+
+if __name__ in ('__main__', '__builtin__', 'builtins'):
+    try:
+        main()
+    except Exception as exception:
+        return_error(exception)
```
Comment (YuvHayun) [inline Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.py:208]:
```suggestion
def main():  # pragma: no cover
    """main function, parses params and runs command functions
    """
    params = demisto.params()
    args = demisto.args()
    command = demisto.command()
    demisto.info(f"Command called {command}")
    demisto.info(f"Args are {args}")
    demisto.info(f"params are {params}")
    client = Client(params)
    try:
      if command == 'quttera-scan-start':
          results = scan_start(client, args)
      elif command == 'quttera-rescan-status':
          results = rescan_status(client, args)
      elif command == 'quttera-scan-report':
          results = scan_report(client, args)
      elif command == 'quttera-report-malware':
          results = report_malware(client, args)
      elif command == 'quttera-status-blacklist':
          results = status_blacklist(client, args)
      elif command == 'test-module':
          results = test_module(client)
      else:
            raise ValueError(f'Command {command} is not implemented in this integration')
      return_results(results)
    except Exception as e:
        return_error(f'Failed to execute {command} command.\nError:\n{str(e)}')


if __name__ in ('__main__', '__builtin__', 'builtins'):
        main()
```

let's put the try and except inside the main only for the party where we ran the commands.

File: Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/README.md [hunk 17]
```diff
@@ -0,0 +1,140 @@
+Quttera Website Malware Scanner
```
Comment (YuvHayun) [inline Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/README.md:1]: Just making sure, did you run the generate-docs command after all your fixes? the readme file is up to date?

**Review by YuvHayun — CHANGES_REQUESTED** [review]
Hey @hayzer done another review after your fixes, job was very good. Left some minor comments for you to fix.
After you'll fix my comments I'll trigger the workflow tests and when those are passed, we'll schedule a short demo with @edik24

**Comment by YuvHayun** [comment]
Hey @hayzer 
I pushed some minor fixes myself to fix indentation issues.
I ran instance tests and everything passed.

The next step is a demo, @edik24 will send you an email to schedule a 30 minutes demo for next week.
Please make sure to fill out the contribution form before the demo.
Also regarding my [question](https://github.com/demisto/content/pull/21350#discussion_r1056955402) please re-run generate-docs if you haven't done that yet.

**Comment by hayzer** [comment]
Hi. I'm traveling. Will be offline till tomorrow. We'll try to create short
video asap.

**Comment by YuvHayun** [comment]
Hey @hayzer, following our demo - few comments to fix:
- Let's add the default URL as the default value of the URL integration param.
- Let's add a comment to tell the user to use the status command after executing the scan command if he wants to get the current execution status.
- Let's edit the human-readable output, no need to add the status code to the human-readable, for commands like the blacklist, we can return free text / a table with three columns - the name, the type (URL, domain, etc..) and blacklisted yes/no.
The integration which we gave as an example in the demo is Microsoft Defender Advanced Threat Protection.
In your attached video please show new readable output after all the mentioned fixes. thx again!

File: Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/README.md [hunk 18]
```diff
@@ -0,0 +1,140 @@
...
+This integration was integrated and tested with version xx of QutteraWebsiteMalwareScanner
```
Comment (YuvHayun) [inline Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/README.md:2]: Replace xx with the tested API version.

File: Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.yml [hunk 19]
```diff
@@ -0,0 +1,157 @@
...
+  type: 4
+- display: Quttera Scanner URL
+  name: base_url
+  required: true
+  defaultvalue: "https://scannerapi.quttera.com"
+  type: 0
+script:
+  type: python
+  subtype: python3
+  dockerimage: demisto/python3:3.10.9.42008
```
Comment (YuvHayun) [inline Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.yml:26]:
```suggestion
  dockerimage: demisto/python3:3.10.9.44472
```

File: Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.yml [hunk 20]
```diff
@@ -144,7 +145,7 @@ script:
       secret: false
       description: The name of the domain to retrieve its blacklist and reputation
     outputs:
-    - contextPath: QutteraWebsiteMalwareScanning.domain
+    - contextPath: QutteraWebsiteMalwareScanning.report
```
Comment (YuvHayun) [inline Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.yml:148]: why did you change this one?

**Review by YuvHayun — CHANGES_REQUESTED** [review]
Hey @hayzer I added a few comments to fix.
Also:
1. Please note your unit tests failed. 
2. Regarding the video - let's make sure to include commands where you changed the readable output (where we added more informational msgs and tables), and also show the cases where the execution fails (like in status_blacklist where you're expecting 404) and context output 

Let me know when you're ready for rereview.

File: Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.yml (hunk 20 above)
Comment (hayzer) [reply Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.yml:148]: Solved

File: Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.yml (hunk 19 above)
Comment (hayzer) [reply Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.yml]: Solved

File: Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/README.md (hunk 18 above)
Comment (hayzer) [reply Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/README.md]: Fixed

**Comment by hayzer** [comment]
Hi Yuval,

Can you take a look, please? I think I fixed most of the things.

Best regards,
Thomas

**Comment by YuvHayun** [comment]
Hey @hayzer, the fixes looks great! the only thing that is missing now is the video.
Let's make sure to include commands where you changed the readable output (where we added more informational msgs and tables), and also show the cases where the execution fails (like in status_blacklist where you're expecting 404) and context output.
Please try to record a quick video asap and I'll approve and merge the issue.

**Comment by YuvHayun** [comment]
@hayzer I got your video sent via mail.
looks great!
merging for internal PR where i'll also fix the last validation.
You should be able to see the pack in the marketplace tomorrow evening.
Thx a lot for your contribution!
